svc.go

```diff
@@ -48,9 +48,10 @@ var (
 
 // AutoNATService provides NAT autodetection services to other peers
 type AutoNATService struct {
-	ctx    context.Context
-	h      host.Host
-	dialer host.Host
+	ctx         context.Context
+	h           host.Host
+	dialer      network.Dialer
+	dialerStore peerstore.Peerstore
```

Comment:
We should just use the one one the host.

Reply:
Ah, now I see. We still don't need the separate peerstore if we use the `network.Network` interface.